src/remote/addr/tests.rs

```diff
@@ -188,3 +191,158 @@ impl Handler<ClusterLog> for OwnListenerGossipIntroduction {
         }
     }
 }
+
+#[derive(RemoteMessage, Serialize, Deserialize)]
+struct ResponseTestMessage {}
+
+#[derive(RemoteMessage, Serialize, Deserialize)]
+struct ResponseTest(pub String);
+
+#[derive(Debug)]
+enum InternalError {
+    Timeout,
+}
+
+#[derive(Message)]
+#[rtype(result = "Result<(), InternalError>")]
+struct InternalTrigger();
+
+#[derive(RemoteActor)]
+#[remote_messages(ResponseTestMessage, ResponseTest)]
+struct SendTestActor {
+    members: Arc<Mutex<Vec<RemoteAddr>>>,
+    subscription_senders: HashMap<TypeId, Sender<Box<dyn Any + Send + 'static>>>,
+}
+
+impl SendTestActor {
+    pub fn new() -> Self {
+        Self {
+            members: Arc::new(Mutex::new(vec![])),
+            subscription_senders: HashMap::new(),
+        }
+    }
+}
+
+impl Actor for SendTestActor {
+    type Context = Context<Self>;
+
+    fn started(&mut self, ctx: &mut Context<Self>) {
+        self.register(ctx.address().recipient());
+        self.subscribe_system_async::<ClusterLog>(ctx);
+    }
+}
+
+impl Handler<InternalTrigger> for SendTestActor {
+    type Result = ResponseFuture<Result<(), InternalError>>;
+    fn handle(&mut self, _: InternalTrigger, ctx: &mut Context<Self>) -> Self::Result {
+        let self_addr = ctx.address();
+        let members = self.members.clone();
+        Box::pin(async move {
+            for remote_addr in members.lock().unwrap().iter() {
+                let future = remote_addr
+                    .send::<_, ResponseTest, _>(ResponseTestMessage {}, &self_addr.clone());
+                let res = tokio::time::timeout(Duration::from_secs(2), future)
+                    .await
+                    .unwrap();
+                res.map_err(|_| InternalError::Timeout)?;
+            }
+            Ok(())
+        })
+    }
+}
+
+impl Handler<ResponseTestMessage> for SendTestActor {
+    type Result = ResponseFuture<()>;
+    fn handle(&mut self, _: ResponseTestMessage, _: &mut Context<Self>) -> Self::Result {
+        let members = self.members.clone();
+        Box::pin(async move {
+            let members = members.lock().unwrap();
+            for remote_addr in members.iter() {
+                remote_addr.do_send(ResponseTest("Hello from the other side".to_string()));
+            }
+        })
+    }
+}
+
+impl Handler<ResponseSubscribe> for SendTestActor {
```

Comment:
This can be probably hidden behind a macro for convenience. I think this is always the same, right?

Reply:
I think it would be similar in 95% of cases, but we need to introduce additional trait with method like `subscribe_to_response` in order to create handy macro